src/appshell/view/dockwindow/dockpageview.cpp

```diff
@@ -213,6 +232,130 @@ DockPanelView* DockPageView::findPanelForTab(const DockPanelView* tab) const
     return !panels.isEmpty() ? panels.first() : nullptr;
 }
 
+void DockPageView::reorderSections()
+{
+    //! NOTE: In some cases, such as setting visible true,
+    //! it is necessary to give the UI time to render the content, so we will add a delay.
+    QTimer::singleShot(2000, this, [this](){
+        doReorderSections();
+    });
+}
+
+void DockPageView::doReorderSections()
+{
+    QList<DockBase*> docks = allDocks();
+    QList<DockBase*> _docks;
+
+    for (DockBase* dock: docks) {
+        if (dock->contentNavigationPanel() && dock->isVisible()) {
+            _docks.append(dock);
+        }
+    }
+
+    QHash<ui::INavigationSection*, QList<DockBase*> > orderedSections = reorderDocksNavigationSections(_docks);
+
+    for (QList<DockBase*>& panels : orderedSections.values()) {
+        reorderNavigationSectionPanels(panels);
+    }
+}
+
+QHash<mu::ui::INavigationSection*, QList<DockBase*> > DockPageView::reorderDocksNavigationSections(QList<DockBase*>& docks)
+{
+    std::sort(docks.begin(), docks.end(), [this](DockBase* dock1, DockBase* dock2) {
+        if (!dock1->contentNavigationPanel() || !dock2->contentNavigationPanel()) {
+            return false;
+        }
+
+        QPoint dock1Pos = dock1->globalPosition();
+        QPoint dock2Pos = dock2->globalPosition();
+
+        if (dock1Pos.y() == dock2Pos.y()) {
+            if (dock1 == centralDock()) {
+                return true;
+            } else if (dock2 == centralDock()) {
+                return false;
+            }
+
+            return dock1Pos.x() < dock2Pos.x();
+        }
+
+        return dock1Pos.y() < dock2Pos.y();
+    });
+
+    int i = 0;
+    QHash<ui::INavigationSection*, QList<DockBase*> > orderedSections;
+    for (DockBase* dock: docks) {
+        auto panel = dock->contentNavigationPanel();
+        if (!panel) {
+            continue;
+        }
+
+        ui::INavigationSection* section = panel ? panel->section() : nullptr;
+        if (section && !orderedSections.contains(section)) {
+            auto index = section->index();
+            index.setOrder(i++);
+            section->setIndex(index);
+        }
+
+        orderedSections[section] << dock;
+    }
+
+    return orderedSections;
+}
+
+void DockPageView::reorderNavigationSectionPanels(QList<DockBase*>& sectionDocks)
+{
+    std::sort(sectionDocks.begin(), sectionDocks.end(), [](DockBase* dock1, DockBase* dock2) {
+        if (!dock1->contentNavigationPanel() || !dock2->contentNavigationPanel()) {
+            return false;
+        }
+
+        QPoint dock1Pos = dock1->globalPosition();
+        QPoint dock2Pos = dock2->globalPosition();
+
+        if (dock1->floating() && dock2->floating()) {
+            return dock1Pos.x() < dock2Pos.x();
+        } else if (dock1->floating()) {
+            return false;
+        } else if (dock1->floating()) {
```

Comment:
The condition in this else if is the same as in the previous one. Looks like a typo